dbfit-java/core/src/main/java/dbfit/api/DbStoredProcedureCall.java

```diff
@@ -51,7 +51,12 @@ void bindParametersTo(StatementExecution cs) throws SQLException {
     }
 
     public StatementExecution toStatementExecution() throws SQLException {
-        StatementExecution cs = new StatementExecution(this.environment.getConnection().prepareCall(toSqlString()));
+        StatementExecution cs;
+        if (isFunction()) {
+            cs = environment.createFunctionStatementExecution(this.environment.getConnection().prepareCall(toSqlString()), true);
```

Comment:
Do you mean some refactoring or re-organsing what's already there?

Something like: -
```Java
    public StatementExecution toStatementExecution() throws SQLException {
        String sql = toSqlString();
        PreparedStatement ps = this.environment.getConnection().prepareCall(sql);
        StatementExecution cs;
        if (isFunction()) {
            cs = environment.createFunctionStatementExecution(ps, true);
        } else {
            cs = environment.createStatementExecution(ps, true);
        }
        bindParametersTo(cs);
        return cs;
    }